docs/help/contributor/style-guide/style-and-formatting.md

```diff
@@ -30,13 +30,36 @@
 
 ## Links
 
-### Make links descriptive
+### Describe what the link targets
 
 |:white_check_mark: Correct                                     |:no_entry: Incorrect
 |---------------------------------------|------
 |For an explanation of what makes a good hyperlink, see this [this article](https://medium.com/@heyoka/Correctnt-use-click-here-f32f445d1021).| See this article [here](https://medium.com/@heyoka/Correctnt-use-click-here-f32f445d1021).
 
-<!-- Add rule here, see https://github.com/knative/docs/issues/4034 -->
+### Write links in Markdown, not HTML
+
+|:white_check_mark: Correct                                     |:no_entry: Incorrect|
+|---------------------------------------|------|
+|`[Kafka Broker](../kafka-broker/README.md)`|`<a href="../kafka-broker/README.md">Kafka Broker</a>`|
+|`[Kafka Broker](../kafka-broker/README.md){target=_blank}`|`<a href="../kafka-broker/README.md" target="_blank">Kafka Broker</a>`|
+
+### Link to files, not folders
+
+|:white_check_mark: Correct                                     |:no_entry: Incorrect|
+|---------------------------------------|------|
+|`[Kafka Broker](../kafka-broker/README.md)`|`[Kafka Broker](../kafka-broker)`|
+
+### Ensure the letter case is correct
+
+|:white_check_mark: Correct                                     |:no_entry: Incorrect|
+|---------------------------------------|------|
+|`[Kafka Broker](../kafka-broker/README.md)`|`[Kafka Broker](../kafka-broker/readme.md)`|
+
+### Include the .md extension in internal links
```

Comment:
I would put this section just above "Link to files, not folders" because it's the main rule for internal linking. The other rules (Link to files, not folders and Using correct case) flow from this main rule.

Reply:
Sure, will do.